core/data/fetch.py

```diff
@@ -190,28 +170,55 @@ def save_img(row: pd.Series, index) -> Any:
                         if xywhn.numel() > 0:
                             img = yolo_crop(img, xywhn)
                             accepted = True
-                    img = FeatureExtractor.make_square_with_bb(img)
-                    img = img.resize((IMG_SIZE, IMG_SIZE))
+                    img = img.resize((constants['IMG_SIZE'], constants['IMG_SIZE']))
+                    img = img.convert("RGB")
                     img = np.asarray(img)
                     np.save(path, img, allow_pickle=True)
                     out = path
                     self.info(out)
                 except requests.exceptions.Timeout:
-                    self.debug(f"Timeout occurred for index {index}")
+                    self.warning(f"Timeout occurred for index {index}")
                 except requests.exceptions.RequestException as e:
-                    self.debug(f"Error occurred: {e}")
+                    self.warning(f"Error occurred: {e}")
+                except urllib3.exceptions.ReadTimeoutError:
+                    self.warning(f"Read timed out on {index}")
                 except ValueError as e:
-                    self.error(f"Image name not applicable: {e}")
-                    raise e
+                    self.error(f"Image name not applicable: {e}", stack_info=True, exc_info=True)
+                    return out, accepted
                 except OSError as e:
-                    self.error(f"Could not save file: {e}")
-                    raise e
+                    self.error(f"Could not save file: {e}", stack_info=True, exc_info=True)
+                    return out, accepted
                 except Exception as e:
-                    self.error(f"Unknown error: {e}")
-                    raise e
+                    self.error(f"Unknown error: {e}", stack_info=True, exc_info=True)
+                    return out, accepted
             else:
-                out = path
-                accepted = True
+                try:
+                    out = row['path']
+                    if pd.isna(out):
+                        raise KeyError
```

Comment:
I don't understand what's happening here with the exceptions

Reply:
if row doesnt have a path column it'll throw a keyerror. Procedure when pd.isna(out) is true and when handling exception is same, so raise same type of error.